### PR TITLE
lib: Add Sorter

### PR DESCRIPTION
**Pull request by robryanx**

### CODE REVIEW

File: runtime/jaktlib/sorter.jakt [hunk 1]
```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2022, Rob Ryan <rob@robryan.me>
+//
+// SPDX-License-Identifier: BSD-2-Clause
+
+enum SortType {
+    Asc
+    Desc
```
Comment (networkException) [inline runtime/jaktlib/sorter.jakt:7]: I don't think there's a reason to abbreviate those

```suggestion
    Ascending
    Descending
```

**Review by networkException — CHANGES_REQUESTED** [review]

**Comment by alimpfard** [comment]
Not going one way or another yet, but future impl could look nicer imo:
```
trait Sortable {
    function sort(mut this, direction: SortDirection) -> void
    function sorted(this, direction: SortDirection) -> Self { ... } // Maybe
}

type Array implements(Sortable) {
    function sort(mut this, direction: SortDirection) -> void { your_impl_here }
}

mut foo = blah.blah()
foo.sort()
foo.sorted()
```

though this would depend on
- extension trait impls on imported classes (Array isn't implemented in Jakt...) OR implementing the method in C++
- Self type in traits (can merge now if needed)
- figuring out the hierarchy of these traits (copy Rust probably lol)

**Comment by robryanx** [comment]
> 



> Not going one way or another yet, but future impl could look nicer imo:
> 
> ```
> trait Sortable {
>     function sort(mut this, direction: SortDirection) -> void
>     function sorted(this, direction: SortDirection) -> Self { ... } // Maybe
> }
> 
> type Array implements(Sortable) {
>     function sort(mut this, direction: SortDirection) -> void { your_impl_here }
> }
> 
> mut foo = blah.blah()
> foo.sort()
> foo.sorted()
> ```
> 
> though this would depend on
> 
>     * extension trait impls on imported classes (Array isn't implemented in Jakt...) OR implementing the method in C++
> 
>     * Self type in traits (can merge now if needed)
> 
>     * figuring out the hierarchy of these traits (copy Rust probably lol)

Yeah, I was trying to figure this out recently for enumerate. I think certainly the trait impls on imported classes will be nice to have to avoid the original issue of an ever expanding prelude and runtime.

File: runtime/jaktlib/sorter.jakt (hunk 1 above)
Comment (robryanx) [reply runtime/jaktlib/sorter.jakt]: Yeah that makes sense, this isn't sql. Updated it.

**Pull request closed without merging**